wikitextprocessor/dumpparser.py

```diff
@@ -180,12 +182,44 @@ def overwrite_pages(
     ctx.db_conn.commit()
     return False
 
+def path_is_on_windows_partition(path: Path) -> bool:
+    """
+    Return True if the path is on an exFAT or NTFS partition.
+    """
+    path_matching_fstypes = [
+        part.fstype for part in disk_partitions() 
+        if str(path.resolve()).startswith(part.mountpoint)
+    ]
+    path_fstype = sorted(path_matching_fstypes, key=len)[-1]
+    return path_fstype == "exfat" or path_fstype == "fuseblk"
```

Comment:
Why only check the last matching file system? And I think "fuseblk" is what NTFS partition displays on Linux but would certainly shows a different value on Windows, maybe just "ntfs".

Reply:
It's looking for a best match in case you've got nested mounts based on this https://stackoverflow.com/a/35291824

Reply:
That's different, here the `path_fstype` variable comes from a list of file systems types sorted by string length not mount point path.

Reply:
You're correct, I just saw the post just beneath that (by bgav) and assumed. Programming is hard, debugging is even harder.

Reply:
@xxyzz @kristian-clausal thanks for catching those mistakes, I've updated the PR to fix them. I also added tests, which really I should have done the first time around. I thought this would be simple but clearly I was wrong.

Reply:
> I thought this would be simple but clearly I was wrong.

I see you've read our motto.